backend/src/types.ts

```diff
@@ -15,6 +15,11 @@ export type DashboardConfig = K8sResourceCommon & {
       disableAppLauncher: boolean;
       disableUserManagement: boolean;
     };
+    groupsConfig?: {
+      [key: string]: string;
```

Comment:
Remove this -- anyone who wants to add to our CR can add it to the type. This will lead to weaker TS.

Reply:
Ah I noticed you did this because of TS -- what you should do is gate the usage by a check -- we don't _want_ to do it if it's not a valid mapping name. The CRD will stop us if it does not exist, but we shouldn't have code that does unexpected things -- makes it harder to maintain and opens us up to bugs.

Just check the value of `groupName` before using it:
```
    if (groupName === 'adminGroups' || groupName === 'allowedGroups') {
      const userGroup = dashCR.groupsConfig[groupName];
```